Sources/Core/Subject/Subject.swift

```diff
@@ -239,19 +247,13 @@ class Subject : NSObject {
         }
     }
 
-    /// Creates a subject which optionally adds platform and geolocation pairs.
-    /// - Parameters:
-    ///   - platformContext: Whether to enable the platform context.
-    ///   - geoContext: Whether to enabled the geolocation context.
-    /// - Returns: A new SPSubject.
-    convenience init(platformContext: Bool = false, andGeoContext geoContext: Bool = false) {
-        self.init(platformContext: platformContext, geoLocationContext: geoContext, subjectConfiguration: nil)
-    }
-
-    /// @warning Internal method - do not use in production
-    init(platformContext: Bool, geoLocationContext geoContext: Bool, subjectConfiguration config: SubjectConfiguration?) {
+    init(platformContext: Bool = false,
+         platformContextProperties: [PlatformContextProperty]? = nil,
+         geoLocationContext geoContext: Bool = false,
+         subjectConfiguration config: SubjectConfiguration? = nil) {
+        self.platformContextManager = PlatformContext(platformContextProperties: platformContextProperties)
         super.init()
-        
+        self.platformContextProperties = platformContextProperties
         self.platformContext = platformContext
         geoLocationContext = geoContext
```

Comment:
it's because in the other properties, the name of the instance variable is the same as the name of the local variable, so we need to use `self.platformContext` to distinguish from the `platformContext` local variable. But the `geoContext` is named differently in both places.